src/abstractdataframe/iteration.jl

```diff
@@ -20,6 +20,13 @@ end
 Base.summary(dfrs::DataFrameRows) = "$(length(dfrs))-element DataFrameRows"
 Base.summary(io::IO, dfrs::DataFrameRows) = print(io, summary(dfrs))
 
+# Trying to iterate over a DataFrame directly (for row in df) tends to
+# be one of the first things a beginner does. Allowing a MethodError
+# to be raised is more "correct", but an explanation of the
+# appropriate function for iterating rows is more friendly.
+Base.iterate(::AbstractDataFrame) =
+    error("AbstractDataFrame is not iterable. Use eachrow(adf) to iterate rows.")
```

Comment:
maybe also add or `eachcol(df) to iterate columns`?
(depending on the background people might expect the iteration to be per-row or per-column)

Reply:
Very good point. The internet connection I'm currently on blocks port 22, so I can't push a fix right now, but I'll try to remember to do so later. Or you can just fix it when merging; whichever works.

Reply:
```suggestion
    error("AbstractDataFrame is not iterable. Use eachrow(df) to get a row iterator" *
          " or eachcol(df) to get a column iterator")
```